### PR TITLE
[MIGRATIONS DA-3142] NPH Intake API

### PR DESCRIPTION
**Pull request by dtharpeuno**
## Resolves *[ticket DA-3142]*
https://precisionmedicineinitiative.atlassian.net/browse/DA-3142

## Description of changes/additions
Adding general rules and scaffolding for adding events utilizing the NPH intake service using example payloads provided by RTI 

## Tests
- [x] unit tests

### CODE REVIEW

File: rdr_service/api/nph_intake_api.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+            return date_time
+
+        except KeyError as e:
+            raise BadRequest(f'KeyError on authored time lookup: {e} bundle_id: {self.bundle_identifier}')
+
+    @auth_required([RTI, RDR])
+    def post(self):
+        intake_payload = request.get_json(force=True)
+        intake_payload = [intake_payload] if type(intake_payload) is not list else intake_payload
```
Comment (jmtrujillo75) [inline rdr_service/api/nph_intake_api.py:137]: Please use `intake_payload = [intake_payload] if isinstance(intake_payload, list) else intake_payload`

**Comment by dtharpeuno** [comment]
It’s the same thing. Type works fine

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: rdr_service/api/nph_intake_api.py (hunk 1 above)
Comment (dtharpeuno) [reply rdr_service/api/nph_intake_api.py:137]: Type works fine for this.

Comment (jmtrujillo75) [reply rdr_service/api/nph_intake_api.py:137]: `isinstance` is more performant in python versions < 3.11. This is fixed in `3.11` AFAIK.

Comment (dtharpeuno) [reply rdr_service/api/nph_intake_api.py:137]: Performance shouldn't be an issue in checking if something is a list or not. Thanks

File: rdr_service/api/nph_intake_api.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        event_dao_instance_items = {k: v for k, v in self.__dict__.items() if 'event_dao' in k}
+        for activity in self.current_activities:
+            event_dao_map[f'{activity.name.lower()}'] = event_dao_instance_items[
+                f'nph_{activity.name.lower()}_event_dao']
+        return event_dao_map
+
+    @classmethod
+    def extract_participant_id(cls, participant_obj: dict) -> str:
+        participant_id = participant_obj['resource']['identifier'][0]['value']
+        participant_id = participant_id.split('/')[-1]
```
Comment (j-kanuch) [inline rdr_service/api/nph_intake_api.py]: The participant IDs we receive have a `1000` appended to the front of the ID the same way in AoU we have the `P` character. Can you add a slice here to remove the 1000?

Comment (dtharpeuno) [reply rdr_service/api/nph_intake_api.py]: Done 
https://github.com/all-of-us/raw-data-repository/pull/3344/commits/7ed37ffeb96955d88dd24ae291c0820bca9b36d4

**Review by j-kanuch — APPROVED** [review]

**Review by kenny-skaggs — APPROVED** [review]

**Pull request merged**